isotovideo

```diff
@@ -31,23 +31,67 @@ ANSI_COLORS_DISABLED or NO_COLOR can be set to any value to disable colors.
 
 Show the current program version and test API version
 
+=item B<-e, -?, --exit-status-from-test-results>
+
+isotovideo will exit with status 1 when a test module fails.
+
 =item B<-h, -?, --help>
 
 Show this help.
 
-=head1 TEST PARAMETER
+=back
+
+=head1 TEST PARAMETERS
 
 All additional command line arguments specified in the C<key=value> format are
 parsed as test parameters which take precedence over the settings in the
 vars.json file. Lower case key names are transformed into upper case
 automatically for convenience.
 
+=head1 EXIT CODES
+
+=over 4
+
+=item B<0 - SUCCESS>
+
+Test ran successfully. Bear in mind that if any of the test modules scheduled
+failed the exit code I<will> be 0.
+
+C<isotovideo> can return non-zero exit codes on test module failures with the
+C<--exit-status-from-test-results> flag.
+
+=item B<1 - ERROR>
+
+An error ocurred during the test execution related to a test backend failure.
+
+=item B<100 - NO TEST MODULES SCHEDULED>
+
+No test module was scheduled.
+
+This exit code is reported when invoked with the C<--exit-status-from-test-results> flag.
+
+
+=item B<101 - TEST MODULES FAILED>
+
+A test module failed.
```

Comment:
"failed" is too strict here considering all the different results a test module can take, see http://open.qa/docs/#_jobs . You can avoid that with negation:

```suggestion
At least one test module did not end with result "passed" or "softfailed".
```

Adjust the header accordingly. I don't know if we define the possible result states in os-autoinst or if we actually only have that definition in the context of openQA

Reply:
Current code suggests: `At least one test module did not end with result "ok"` instead.

Reply:
There's now support for `softfail` on test module result :tada: